databricks/koalas/typedef/typehints.py

```diff
@@ -324,86 +329,134 @@ def infer_return_type(f) -> Union[SeriesType, DataFrameType, ScalarType, Unknown
     """
```

Comment:
Would it be helpful to add a docstring for `infer_return_type` in order to explain return values? For example:

```
Return a type class, which is furthermore understood by Spark.
This type class indicates both dtypes(a pandas only dtype object or a numpy dtype object) and its corresponding Spark DataType.
```